sw/device/lib/runtime/ibex.h

```diff
@@ -27,15 +27,28 @@ extern const size_t kIbexClockFreqHz;
  *
  * @param cycles the number of cycles to burn.
  */
-inline void ibex_busy_loop(size_t cycles) {
-  size_t out;  // Used to create an inout parameter below.
+inline void ibex_busy_loop(uint64_t cycles) {
+  // Round down |cycles| to closest multiple of 8
```

Comment:
If we want the behavior of spinning at least for `cycles` cycles, then this needs to round up.

Reply:
Sure, I rounded it down to keep things simple (since rounding up would require range checking to prevent overflow) but I will change this part as you suggested.

Reply:
Using the cycle counter also resolved this issue.